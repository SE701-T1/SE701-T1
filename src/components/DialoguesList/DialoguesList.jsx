/* eslint-disable react/jsx-props-no-spreading */
import * as React from 'react';

import styles from './DialoguesList.module.css';

export default function DialoguesList({ messages }) {
  const messagesEndRef = React.useRef(null);

  const scrollToBottom = () => {
    messagesEndRef.current?.scrollIntoView({ behavior: 'smooth' });
  };

  React.useEffect(() => {
    scrollToBottom();
  }, [messages]);

  return (
    <div className={styles.root}>
      {messages.map((component, key) => (
        // eslint-disable-next-line react/no-array-index-key
        <div key={key}>{component}</div>
      ))}
      <div className={styles.bottom} ref={messagesEndRef} />
    </div>
  );
}
