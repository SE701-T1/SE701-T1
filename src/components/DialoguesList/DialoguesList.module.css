/* will need to change depending on the rest of the screen */
.root {
  position: absolute;
  bottom: 0;
  width: 800px;
  display: inline-block;
  max-height: 800px;
}

.bottom {
  padding-top: 10px;
}
