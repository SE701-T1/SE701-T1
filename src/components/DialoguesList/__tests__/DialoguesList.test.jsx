import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DialoguesList from '../DialoguesList';

const mockMessage = [
  {
    sender: 'id3333',
    text: 'random message',
    contentType: 'text',
  },
  {
    sender: 'id5678',
    text: 'https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512',
    contentType: 'image',
  },
];

// used for testing automatic scrolling
const scrollIntoViewMock = jest.fn();
window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;

describe('DialoguesList', () => {
  test('display message sent to me', () => {
    render(
      <DialoguesList
        messages={mockMessage}
        currentUser="id5678"
        senderAvatar="https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512"
      />,
    );
    expect(screen.getByText('random message')).toBeInTheDocument();
  });

  test('display image message sent by me', () => {
    render(
      <DialoguesList
        messages={mockMessage}
        currentUser="id5678"
        senderAvatar="https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512"
      />,
    );
    const imageMessage = screen.getByAltText('imageMessage');
    expect(imageMessage).toHaveAttribute(
      'src',
      'https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512',
    );
  });

  it('check that automatic scrolling is done upon new message', () => {
    render(
      <DialoguesList
        messages={mockMessage}
        currentUser="id5678"
        senderAvatar="https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512"
      />,
    );

    expect(scrollIntoViewMock).toBeCalled();
  });

  // snapshot test
  it('produces correct snapshot of multiple messages', () => {
    const tree = renderer
      .create(
        <DialoguesList
          messages={mockMessage}
          currentUser="id5678"
          senderAvatar="https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
