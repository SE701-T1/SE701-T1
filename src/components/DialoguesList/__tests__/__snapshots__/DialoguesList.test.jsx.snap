// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialoguesList produces correct snapshot of multiple messages 1`] = `
<div>
  <div
    className="message-container bubble-from-left"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular css-1u8q9pb-MuiAvatar-root"
    >
      <img
        alt="id3333-image"
        className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512"
      />
    </div>
    <div
      className="bubble others bubble-text"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        random message
      </p>
    </div>
  </div>
  <div
    className="message-container bubble-from-right"
  >
    <div
      className="bubble me bubble-img"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        <img
          alt="imageMessage"
          className="content-img"
          src="https://play-lh.googleusercontent.com/zepWGavYYErAIBXFZb6OT14I6b-m4TyaG3yjqZy6Hnsmi64vL3upQ3KUsV6Wnsm-e9M=w512"
        />
      </p>
    </div>
  </div>
  <div
    className="bottom"
  />
</div>
`;
